processing/src/main/java/org/apache/druid/query/filter/InDimFilter.java

```diff
@@ -382,7 +386,7 @@ public boolean equals(Object o)
   @Override
   public int hashCode()
   {
-    return Objects.hash(values, dimension, extractionFn, filterTuning);
+    return Objects.hash(values.size(), dimension, extractionFn, filterTuning);
```

Comment:
Maybe use the size and the first few values?

It's easy to imagine situations where the extra collisions from only checking size are a problem, and it's tough to imagine situations where the perf impact of adding[the first few values is going to be big. So it seems like a good idea.

Please also include a comment about the rationale for the nonstandard hashCode impl. It'd be good to link to this PR.

Reply:
It's interesting though that `values` itself is a HashSet being passed to InDimFilter which would mean hash code is evaluated for all the elements in the set. But that penalty for constructing `values` doesn't show up in the graph. is the full flame graph available to look further? 
I can see in one place where multiple `InDimFilter` are created with the same `values`.  Maybe that's the part responsible for perf penalty. If there is a `Set` type that remembers its hashCode, using such type for `values` could be more beneficial.

Reply:
@abhishekagarwal87 Good point... I'm not sure why the construction time doesn't show up. I'll check if there is a set that memoizes it's hashcode as the set is being constructed.

Reply:
ImmutableSet computes its hashcode as it is built and then caches it.